tidb/src/tidb/bank.clj

```diff
@@ -91,9 +115,15 @@
                     (if (:update-in-place test)
                       (do (c/execute! c ["update accounts set balance = balance - ? where id = ?" amount from])
                           (c/execute! c ["update accounts set balance = balance + ? where id = ?" amount to])
+                          (when (:test-foreign-key test) (with-txn op [c conn {:isolation (util/isolation-level test)
+                            :before-hook (partial c/rand-init-txn! test conn)}]
+                            (insert-bank-record! c {:from from :to to :amount amount})))
```

Comment:
```suggestion
                          (when (:test-foreign-key test)
                            (insert-bank-record! c {:from from :to to :amount amount}))
```

Since all ops are handled within `with-txn` at L81, we do not need an another `with-txn` here.